src/client/callbacks.rs

```diff
@@ -313,23 +314,27 @@ where
     pub unsafe fn register_callbacks(b: &mut Box<Self>) -> Result<(), Error> {
         let data_ptr = CallbackContext::raw(b);
         let client = b.client.raw();
-        j::jack_set_thread_init_callback(client, Some(thread_init_callback::<N, P>), data_ptr);
-        j::jack_on_info_shutdown(client, Some(shutdown::<N, P>), data_ptr);
-        j::jack_set_process_callback(client, Some(process::<N, P>), data_ptr);
-        j::jack_set_freewheel_callback(client, Some(freewheel::<N, P>), data_ptr);
-        j::jack_set_buffer_size_callback(client, Some(buffer_size::<N, P>), data_ptr);
-        j::jack_set_sample_rate_callback(client, Some(sample_rate::<N, P>), data_ptr);
-        j::jack_set_client_registration_callback(
+        (LIB.jack_set_thread_init_callback)(client, Some(thread_init_callback::<N, P>), data_ptr);
```

Comment:
I'm confused by this. The dlib documentation says you need to use the `ffi_dispatch!` macro for this.

Reply:
You don't really need it. I could update all the callers to enable static and dynamic dispatch.

Reply:
Full disclosure: I couldn't get the macro to work.

Other thing the `dlib` library seemed not to work well is specifying both statics and functions in the same definition. Even the README's example does not compile.

Reply:
Not being able to link directly defeats the purpose of using dlopen instead of using libloading directly.

Reply:
Fixed `ffi_dispatch!`. The issues I ran into were lack of support for trailing commas and no support for functions that don't have any arguments. The latter cases were handled manually, luckily there were only a couple of them.

Reply:
Great, thanks for taking care of that. I suggest reporting those issues upstream.